File: BaekjoonWithSwift/BaekjoonWithSwift/정렬2/시리얼번호.swift
//
//  시리얼번호.swift
//  BaekjoonWithSwift
//
//  Created by kyungmin on 2024/01/10.
//

import Foundation

func 시리얼번호() {
    let N = Int(readLine()!)!
    var serial = [(Int, Int, String)]()
    
    for _ in 0..<N {
        let st = readLine()!
        var sum = 0
        for i in st {
            let asciiNum = UnicodeScalar(String(i))!.value
            if asciiNum >= 49 && asciiNum <= 57 {
                sum += Int(String(i))!
            }
        }
        serial.append((st.count, sum, st))
    }
    
    serial.sort { ($0.0, $0.1, $0.2) < ($1.0, $1.1, $1.2) }
    
    for (_, _, s) in serial {
        print(s)
    }
}

File: BaekJoon/정렬 2/시리얼 번호.py
"""
author: kyungmin
date: 24.1.10
title: 시리얼 번호
time: 30분
"""
import sys
input = sys.stdin.readline

N = int(input().rstrip())

serial = []
judge = dict()

for _ in range(N):
    S = input().rstrip()
    serial.append(S)
    sum = 0
    for i in S:
        if 49 <= ord(i) <= 57:
            sum += int(i)
    judge[S] = sum


serial.sort(key=lambda x: (len(x), judge[x], x))
print(*serial, sep = "\n")
